docs/developer/feeds.rst

```diff
@@ -4,122 +4,217 @@
+registered feeds, and subscribing to other agent's feeds.  The feed system adds
+some structure to the base pub/sub layer by adding features such as
+data-caching, data and feed-name verification, and aggregation behavior
+customization.
 
 
 Registering Feeds
 ------------------------
 
-An ``OCSAgent`` can register a feed by calling::
-
-    agent.register_feed(feed_name, **kwargs)
-
-where ``**kwargs`` are the keyword arguments passed to the Feed's init method
-specified in the above API. This feed will publish data to the unique WAMP
-address ``agent_address.feeds.feed_name``. For instance, in the LS240_Agent,
-readout data is published to a registered feed called *temperatures*.
-In this case, if I run LS240_Agent with instance-id ``thermo1``,
-the agent_address will be ``observatory.thermo1`` and the feed's address
-will be ``observatory.thermo1.feeds.temperatures``.
-
-The dictionary ``agg_params`` is passed to the aggregator, and tells the
-aggregator if and how a feed should be aggregated.
-For now, the only param used is ``"aggregate"`` which determines whether
-the aggregator should subscribe to the feed and write its data to a G3Timestream.
-This is set by default to be False. Eventually, ``agg_params`` will also be able
-to determine how long each frame should be, and the structure of the G3Frame.
-
-Publishing to a Feed
-----------------------
-You can publish data to a feed by calling::
-
-    agent.publish_to_feed(feed_name, message)
-
-This publishes the tuple ``(message, feed_data)``  to the feed's address.
-``message`` is the message passed to the ``publish_to_feed`` function,
-and ``feed_data`` is a dictionary representation of the feed itself.
-This dictionary contains the following fields:
-
+A custom Agent class can register a feed using its ``OCSAgent`` instance. For
+example, if the OCSAgent instance is stored in the ``self.agent`` variable, a
+basic feed can be registered by calling::
 
-==================   =================================================
-agent_address        Address of the agent the feed is registered to.
-feed_name            Name of the feed
-address              Full WAMP address of the feed
-messages             Cache of previous messages
-agg_params           Parameters passed to the Aggregator
-buffered             Specifies if Feed data is buffered
-buffer_time          Specifies how long feed data is buffered,
-                     or how long it should be buffered by the aggregator
-==================   =================================================
+    self.agent.register_feed(feed_name)
 
+The ``register_feed`` function takes a few other key word arguments to customize
+its behavior (See :ref:`OCSAgent Api <ocs_agent_api>` for more details).
+``buffer_time`` will set how long the feed should buffer messages before sending
+over crossbar, and ``max_messages`` will set how many messages are cached.
 
+Feed_name Rules
```

Comment:
Just aesthetically I think it'd be better for the heading to be "Feed Name Rules" or "Feed Naming Rules".